glean-core/src/lib.rs

```diff
@@ -308,22 +334,27 @@ impl Glean {
         self.upload_enabled
     }
 
-    /// Handles the changing of state when upload_enabled changes.
+    /// Handles the changing of state from upload disabled to enabled.
     ///
     /// Should only be called when the state actually changes.
-    /// When disabling, all pending metrics, events and queued pings are cleared.
     ///
-    /// When enabling, the core Glean metrics are recreated.
+    /// The core Glean metrics are recreated.
+    fn on_upload_enabled(&mut self) {
+        self.upload_enabled = true;
+        self.initialize_core_metrics();
+    }
+
+    /// Handles the changing of state from upload enabled to disabled.
     ///
-    /// # Arguments
+    /// Should only be called when the state actually changes.
     ///
-    /// * `flag` - When true, enable metric collection.
-    fn on_change_upload_enabled(&mut self, flag: bool) {
-        if flag {
-            self.initialize_core_metrics();
-        } else {
-            self.clear_metrics();
+    /// All pending metrics, events and queued pings are cleared.
```

Comment:
The docs should call out that it requires the flag to still be true for a deletion request ping to be sent (IMO that could also be a comment on the `if let Err` line below, given that we might not want to document the exact details of the function in its  documentation)